src/nodes.coffee

```diff
@@ -486,8 +486,9 @@ exports.Root = class Root extends Base
   # better not to generate them in the first place, but for now, clean up
   # obvious double-parentheses.
   compileNode: (o) ->
-    o.indent  = if o.bare then '' else TAB
-    o.level   = LEVEL_TOP
+    o.indent    = if o.bare then '' else TAB
+    o.level     = LEVEL_TOP
+    o.compiling = yes
```

Comment:
Introduce a flag on `o` that lets us distinguish (in node class methods shared by AST generation and compile-to-JS) whether we're "compiling" (to JS) as opposed to generating AST